scripts/regclient/utils.lua

```diff
@@ -0,0 +1,36 @@
+local utils = {}
+
+local gcr = "gcr.io"
+
+function utils.parseVersion(v)
+  return string.match(v, "^(%d+)%.(%d+)%.(%d+)(.-)$")
+end
+
+function utils.combine(a, b)
+  -- create a path that looks like a directory, this is supported by gcr
+  local dirPath = a .. "/" .. b
+  if string.sub(dirPath, 1, #gcr) == gcr then
+    return dirPath
+  end
+
```

Comment:
why can't the rest of this just be `return a .. "/" .. string.gsub(b, "/", "-")`

Reply:
because registries allow namespaces, this will be used later for copying from say `docker.io` + `curlimages/curl` which should map to `docker.io/curlimages/curl` not `docker.io/curlimages-curl`